### ISSUE TITLE
Feature: Don't sort styles in react components

### ISSUE THREAD

**Issue by olhapi** [azat-io/eslint-plugin-perfectionist#31]
### What rule do you want to change?

perfectionist/sort-objects

### Describe the problem

Rule `perfectionist/sort-objects` breaks styles when sorting objects passed to `styled` components.

After sorting `borderWidth` will be after `borderBottom` which will rewrite it. Sorting should be smart enough to understand stuff like that (like style lint) or be disabled for styled components (or at least have an option for that).

![image](https://github.com/azat-io/eslint-plugin-perfectionist/assets/4780263/2c8a5c2e-3f56-4c84-996f-20a14056b7b5)

### Code example

```
export default styled('div')(() => ({
    borderRadius: 0,
    borderWidth: 0,
    border: 0,
    borderBottom: hasBorder && `1px solid ${theme.palette.divider}`,
}));
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
This is a very interesting case. I don't think it can be done quickly. But I will definitely think about how it can be solved.

**Comment by olhapi** [comment]
> This is a very interesting case. I don't think it can be done quickly. But I will definitely think about how it can be solved.

and a vary common one I believe, for now I've just disabled object sorting for safety, but would be good to have it enabled for non-styles

otherwise it work pretty good, the code look much better when sorted properly

**Comment by azat-io** [comment]
Added in #44
Will be released soon. See #32.

**Comment by olhapi** [comment]
@azat-io cool stuff, thanks 🎉🎉

**Comment by azat-io** [comment]
Released in [v2.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v2.0.0)
